app/src/main/java/xyz/whereuat/whereuat/ui/views/AddContactDialogFragment.java

```diff
@@ -0,0 +1,119 @@
+
+    @Override
+    public Dialog onCreateDialog(final Bundle saved_instance_state) {
+        final Activity activity = getActivity();
+
+        AlertDialog.Builder builder = new AlertDialog.Builder(activity, R.style.DialogTheme);
+        LayoutInflater inflater = activity.getLayoutInflater();
+        View view = inflater.inflate(R.layout.add_contact_dialog, (ViewGroup) this.getView());
+        builder.setView(view)
+                .setTitle(R.string.add_contact_prompt_title_text)
+                // When clicked, the "Okay" button should add the key location to the database if it
+                // is valid.
+                .setPositiveButton(R.string.okay, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        Dialog d = (Dialog) dialog;
+                        EditText e = (EditText) d.findViewById(R.id.contact_name_input);
+
+                        String name = e.getText().toString();
+
+                        // Make sure the name is not null or empty and the location is not null.
+                        if (!name.isEmpty()) {
+                            addContact(activity, name, getArguments().getString(
+                                    Constants.CONTACT_NUMBER_BUNDLE_EXTRA));
+                        } else {
+                            String err_str = "Invalid contact name";
+                            Toast.makeText(activity, err_str, Toast.LENGTH_SHORT).show();
+                        }
+                    }
+                })
+                // The "Cancel" button shouldn't do anything.
+                .setNegativeButton(R.string.cancel, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {}
+                });
+        return builder.create();
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        final AlertDialog d = (AlertDialog) getDialog();
+
+        EditText e = (EditText) d.findViewById(R.id.contact_name_input);
+        // Automatically show the keyboard when the dialog appears.
+        e.setOnFocusChangeListener(new View.OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View v, boolean hasFocus) {
+                if (hasFocus) {
+                    d.getWindow().setSoftInputMode(
+                            WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_VISIBLE);
+                }
+            }
+        });
+    }
+
+    /**
+     * This function wraps an insert into the pending requests table.
+     *
+     * @param activity the Activity that the fragment comes from
+     * @param name the name of the new contact
+     * @param number the number of the new contact
+     */
+    private void addContact(final Activity activity, final String name, final String number) {
```

Comment:
Can we rename this to make it clear it's also deleting from the pending requests table? Maybe `addContactFromPending` or something